App.js

```diff
@@ -1,26 +1,111 @@
-import React from "react"
-import styled from "styled-components/native"
+import React, { useState } from 'react'
+import { TouchableOpacity, ScrollView } from 'react-native'
+import styled from "styled-components"
+
+import Icon from 'react-native-vector-icons/Feather'
+
+import TodoList from './Components/TodoList'
+
+
+
+export default function App() {
+  const [value, setValue] = useState('')
+  const [todos, setTodos] = useState([])
+
+  addTodo = () => {
+    if (value.length > 0) {
+      setTodos([...todos, { text: value, key: Date.now(), checked: false }])
+      setValue('')
+    }
+  }
+
+  checkTodo = id => {
+    setTodos(
+      todos.map(todo => {
+        if (todo.key === id) todo.checked = !todo.checked
+        return todo
+      })
+    )
+  }
+
+  deleteTodo = id => {
+    setTodos(
+      todos.filter(todo => {
+        if (todo.key !== id) return true
+      })
+    )
```

Comment:
nice, this could also be shortened to
`setTodos(todos.filter(todo => todo.key !== id)`